client/src/composables/fileDrop.ts

```diff
@@ -39,40 +49,53 @@ export function useFileDrop(dropZone, onDrop, solo) {
         document.body,
         "drop",
         (event) => {
-            if (!dragBlocked.value) {
+            if (!dragBlocked) {
                 // prevent the browser from opening the file
                 event.preventDefault();
 
                 if (isFileOverDropZone.value && isFileOverDocument.value) {
-                    unref(onDrop)(event);
+                    unref(onDrop)(event as DragEvent);
                 }
             }
             isFileOverDocument.value = false;
-            dragBlocked.value = false;
+            dragBlocked = false;
+            hasExited = true;
         },
         true
     );
 
-    useEventListener(
-        document.body,
-        "dragend",
-        () => {
-            // reset on drag end
-            isFileOverDocument.value = false;
-            isFileOverDropZone.value = false;
-            dragBlocked.value = false;
-        },
-        true
-    );
+    /** Reset all variables */
+    const reset = () => {
+        isFileOverDocument.value = false;
+        isFileOverDropZone.value = false;
+        dragBlocked = false;
+        hasExited = true;
+    };
+
+    useEventListener(document.body, "dragend", reset, true);
+
+    useEventListener(document.body, "dragleave", async () => {
+        hasExited = true;
+
+        // This event may have been triggered by components
+        // which have not been properly childed to the body yet.
+        // Wait a bit, and check if hasExited is still true.
+        await wait(100);
```

Comment:
No, that would just shedule it to run before the browsers next draw loop. It might delay the function enough, but I wouldn't want to rely on it, as it might depend on the browsers execution order.
Vue nextTick could work though, I can try that. The underlying solution would be the same: to wait for the modals dom state to have settled.
Another solution I considered is checking if the event coordinates overlap with the body, but that would not have worked with native popups, such as the issue on the workflow form. While the waiting solution feels a bit hacky, I think it may be the most robust.

Reply:
Update: Vue.nextTick(); does not work reliably. The modal seems to not child to the body within a tick. Two work for the first modal, but not subsequent ones. I do not know why this is the case, but think it is out of our control.

Reply:
Thanks for checking!